be/src/runtime/stream_load/stream_load_executor.cpp

```diff
@@ -223,54 +224,50 @@ Status StreamLoadExecutor::commit_txn(StreamLoadContext* ctx) {
         request.__isset.txnCommitAttachment = true;
     }
 
-    auto st = commit_txn_internal(request, rpc_timeout_ms, ctx);
-    // service unavailable implies that the exception occurred in network transmission or Thrift RPC framework,
-    // rather than due to slow publish. Therefore, we can utilize the remaining timeout to retry.
-    if (st.is_service_unavailable()) {
-        auto remain_timeout_ms = ctx->load_deadline_sec > 0 ? (ctx->load_deadline_sec - UnixSeconds()) * 1000 : 0;
-        if (remain_timeout_ms > 0) {
-            return commit_txn_internal(request, remain_timeout_ms, ctx);
+    int retry = 0;
+    TLoadTxnCommitResult result;
+    while (true) {
+        RETURN_IF_ERROR(commit_txn_internal(request, rpc_timeout_ms, &result));
+        Status st(result.status);
+        if (st.ok()) {
```

Comment:
what about `st.is_service_unavailable()`

Reply:
there is no service unavailable error now. 
when encountering `TTransportException`, rpc will try again in `thrift_rpc_helper`.